### PR TITLE
Cherry-pick #15497 to 7.x: [Packetbeat] Output ECS-compatible TLS fields

### PR DESCRIPTION
**Pull request by adriansr**
Cherry-pick of PR #15497 to 7.x branch. Original message: 

Packetbeat now outputs TLS fields from ECS 1.3+:
- The additional information not covered by ECS is nested under `tls.detailed`.
- Fields already in ECS are removed from detailed to avoid bloat.
- A new configuration flag `tls.include_detailed_fields` allows to
  toggle the inclusion of extra fields. It's enabled by default.

Caveats:
- Originally it would output the top-level certificate in `tls.server_certificate` and the rest under `tls.server_certificate_chain`. ECS mandates that `tls.server.certificate` and `tls.server.certificate_chain` are mutually exclusive. To avoid confusion, a chain is always generated, even if it consists of a single certificate.
- Same for `tls.client` certificates.
- The behavior of the configuration options `tls.send_certificates` and `tls.include_raw_certificates` has slightly changed.

Non-populated TLS ECS fields:
- `tls.curve`: Not implemented. Requires parsing the server key exchange.
- `tls.server.ja3s`: JA3s is not implemented yet.

### CODE REVIEW

**Comment by adriansr** [comment]
jenkins, test this

File: packetbeat/protos/tls/_meta/fields.yml [hunk 1]
```diff
@@ -352,3 +352,351 @@
...
+    #
+    # Aliases from pre-ECS fields to tls.detailed
+    #
+
+    # Replaced fields
+
+    - name: tls.handshake_completed
+      type: alias
+      path: tls.established
+      migration: true
```
Comment (andrewkroh) [inline packetbeat/protos/tls/_meta/fields.yml]: Do these fields show up in `filebeat export template` by default? I'm just wondering what effect `migration: true` has? Or does `migration.6_to_7.enabled: true` need to be set for them to show up (I don't think we want that to be needed).

Comment (adriansr) [reply packetbeat/protos/tls/_meta/fields.yml]: You're right. I removed it and now they show up without the 6 to 7 flag.

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -42,6 +42,12 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 
 *Packetbeat*
 
+- Add dns.question.subdomain and dns.question.top_level_domain fields. {pull}14578[14578]
```
Comment (andrewkroh) [inline CHANGELOG.next.asciidoc]: Git gave you more than you wanted here.

**Pull request merged**